File: app1_terminada/editar_libro.c
#include "function.h"

int editar_libro(struct struct_libro array_libros[], int arr_size)
{
    char titulo[20];
    char temp;

    printf("\npara editar un libro, ingrese el titulo: ");
    scanf("%c", &temp);
    scanf("%[^\n]%*c", titulo);
    printf("\nEditaremos: %s\n", titulo);

    for (int i = 0; i <= arr_size; i++)
    {
        if (strcmp(titulo, array_libros[i].titulo) == 0)
        {
            int opcion = 0;
            printf("\n---Elija el numero de la opcion que quiere editar---");
            printf("\n1) Cambiar titulo");
            printf("\n2) Cambiar autor");
            printf("\n3) Cambiar anio");
            printf("\n4) Cambiar numero de estante");
            printf("\n5) Cambiar seccion");
            printf("\n6) Cambiar piso");
            printf("\n7) Cambiar edificio");
            printf("\n8) Cambiar sede");
            printf("\n9) Volver al menu");
            printf("\nIngrese numero: ");

            scanf("%d", &opcion);
            if (opcion == 1)
            {
                char cambio_titulo[100];
                printf("\nEl titulo actual del libro es: %s", array_libros[i].titulo);
                printf("\nIngrese el nuevo titulo: ");
                scanf("%c", &temp);
                scanf("%[^\n]%*c", cambio_titulo);
                strcpy(array_libros[i].titulo, cambio_titulo);
                printf("\nNuevo titulo: %s", array_libros[i].titulo);
            }
            else if (opcion == 2)
            {
                char cambio_autor[100];
                printf("\nEl/los autores actuales son: %s", array_libros[i].autor);
                printf("\nIngrese el nuevo autor: ");
                scanf("%c", &temp);
                scanf("%[^\n]%*c", cambio_autor); 
                strcpy(array_libros[i].autor, cambio_autor);
                printf("\nNuevo autor: %s", array_libros[i].autor);
            }
            else if (opcion == 3)
            {
                int cambio_anio;
                printf("\nEl anio actual es: %d", array_libros[i].anio);
                printf("\nIngrese el nuevo anio: ");
                scanf("%c", &temp);
                scanf("%d", &cambio_anio);
                array_libros[i].anio = cambio_anio;
                printf("\nNuevo anio: %d", array_libros[i].anio);
            }
            else if (opcion == 4)
            {
                int cambio_estante;
                printf("\nEl numero de estante actual es: %d", array_libros[i].estante_numero);
                printf("\nIngrese el nuevo numero de estante: ");
                scanf("%c", &temp);
                scanf("%d", &cambio_estante);
                array_libros[i].estante_numero = cambio_estante;
                printf("\nEl libro ahora se encuentra en el estante: %d", array_libros[i].estante_numero);
            }
            else if (opcion == 5)
            {
                char cambio_seccion[10];
                printf("\nEl libro se encuentra actualmente en la seccion de: %s", array_libros[i].estante_seccion);
                printf("\nIngrese la seccion a la cual desea mover el libro: ");
                scanf("%c", &temp);
                scanf("%[^\n]%*c", cambio_seccion);
                strcpy(array_libros[i].estante_seccion, cambio_seccion);
                printf("\nEl libro se encuentra ahora en la seccion: %s", array_libros[i].estante_seccion);
            }
            else if (opcion == 6)
            {
                int cambio_piso;
                printf("\nEl numero del piso actual es: %d", array_libros[i].estante_numero);
                printf("\nIngrese a que piso cambiara el libro: ");
                scanf("%c", &temp);
                scanf("%d", &cambio_piso);
                array_libros[i].piso = cambio_piso;
                printf("\nEl libro se encuentra ahora en el piso: %d", array_libros[i].estante_numero);
            }
            else if (opcion == 7)
            {
                char cambio_edificio[10];
                printf("\nEl libro se encuentra actualmente en el edificio: %s", array_libros[i].edificio);
                printf("\nIngrese a que edificio movera el libro: ");
                scanf("%c", &temp);
                scanf("%[^\n]%*c", cambio_edificio);
                strcpy(array_libros[i].edificio, cambio_edificio);
                printf("\nEl libro se encuentra ahora en el edificio: %s", array_libros[i].edificio);
            }
            else if (opcion == 8)
            {
                char cambio_sede[20];
                printf("\nEl libro se encuentra actualmente en la sede: %s", array_libros[i].sede);
                printf("\nIngrese a que sede (vina o santiago) movera el libro: ");
                scanf("%c", &temp);
                scanf("%[^\n]%*c", cambio_sede);
                strcpy(array_libros[i].sede, cambio_sede);
                printf("\nEl libro se encuentra ahora en la sede: %s", array_libros[i].sede);
            }
            else if (opcion == 9)
            {
                scanf("%c", &temp);
                return 0;
            }
        }
    }

    return 0;
}

File: BenjaApp2/Editarlibro.java

import java.util.*;

public class Editarlibro {

    static Scanner input = new Scanner(System.in);

    public int opcionEditar() {
        int opcionElegida;
        System.out.println("1) Cambiar titulo");
        System.out.println("2) Cambiar autor");
        System.out.println("3) Cambiar anio");
        System.out.println("4) Cambiar numero de estante");
        System.out.println("5) Cambiar seccion");
        System.out.println("6) Cambiar piso");
        System.out.println("7) Cambiar edificio");
        System.out.println("8) Cambiar sede");
        System.out.println("9) Volver al menu");
        System.out.println("Ingrese que quiere editar de el libro:");
        return opcionElegida = input.nextInt();
    }

    public Books editarLibro(List<Books> bookList, String bookToEdit) {
        int opcionElegida = opcionEditar();
        for (Books book : bookList) {
            if (book.getTitulo().equals(bookToEdit)) {
                if (opcionElegida == 1) {
                    System.out.println("Ingrese el titulo nuevo");
                    String temp = input.nextLine();
                    String titulo = input.nextLine();
                    book.setTitulo(titulo);
                }
                if (opcionElegida == 2) {
                    System.out.println("Ingrese el autor nuevo");
                    String temp = input.nextLine();
                    String autor = input.nextLine();
                    book.setAutor(autor);
                }
                if (opcionElegida == 3) {
                    System.out.println("Ingrese el anio nuevo");
                    String temp = input.nextLine();
                    int anio = input.nextInt();
                    book.setAnio(anio);
                }
                if (opcionElegida == 4) {
                    System.out.println("Ingrese el numero de estante nuevo");
                    String temp = input.nextLine();
                    int numeroEstante = input.nextInt();
                    book.setNumeroEstante(numeroEstante);
                }
                if (opcionElegida == 5) {
                    System.out.println("Ingrese la seccion nueva");
                    String temp = input.nextLine();
                    String seccionEstante = input.nextLine();
                    book.setSeccionEstante(seccionEstante);
                }
                if (opcionElegida == 6) {
                    System.out.println("Ingrese el piso nuevo");
                    String temp = input.nextLine();
                    int piso = input.nextInt();
                    book.setPiso(piso);
                }
                if (opcionElegida == 7) {
                    System.out.println("Ingrese el edificio nuevo");
                    String temp = input.nextLine();
                    String edificio = input.nextLine();
                    book.setEdificio(edificio);
                }
                if (opcionElegida == 8) {
                    System.out.println("Ingrese la sede nueva");
                    String temp = input.nextLine();
                    String sede = input.nextLine();
                    book.setSede(sede);
                }
            }
        }
        return null;
    }
}